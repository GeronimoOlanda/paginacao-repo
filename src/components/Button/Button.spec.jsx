import { fireEvent ,render, screen } from "@testing-library/react";
import { Button } from '.';

describe('<Button />', () => {
    it('should render the button with the text "Load more"', () => {
        render(<Button text="Load more" />);
        expect.assertions(1);

        const button = screen.getByRole('button', {name: /load more/i}); 
        expect(button).toHaveAttribute('class', 'button');
    })
    it('should call function on button click', () => {
        const fn = jest.fn();
        render(<Button text="Load more" onClick={fn} />);

        const button = screen.getByRole('button', {name: /load more/i}); 
        fireEvent.click(button); //executa o evento de click no botao
        expect(fn).toHaveBeenCalledTimes(1)
    })
});

// screen utilizado para pegar o botao
// query nao levanta o erro se nao encontrar
// get utilizamos para quando sabemos que o elemento esta na tela
// visando acessibilidade, caso o elemento nao encontrado utilizamos o getbyrole