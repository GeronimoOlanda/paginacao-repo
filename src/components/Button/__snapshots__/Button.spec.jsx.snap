// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> shoulds call function on button click 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
